tests/http/HttpClientTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace go1\rest\tests\http;
+
+use go1\rest\Request;
+use go1\rest\Response;
+use go1\rest\tests\fixtures\FoodCreatedEvent;
+use function json_encode;
+use Psr\Http\Client\ClientInterface;
+use Symfony\Component\HttpClient\CurlHttpClient;
+use Symfony\Component\HttpClient\Psr18Client;
+
+class HttpClientTest extends RestTestCase
+{
+    public function testForwardRequestId()
+    {
+        $rest = $this->rest();
+        $_SERVER['HTTP_X_REQUEST_ID'] = $requestId = 'abcd-1234';
+
+        /**
+         * @var $client Psr18Client
+         */
+        $client = $rest->getContainer()->get(ClientInterface::class);
+
+        $httpClient = call_user_func(function () use ($client) {
```

Comment:
We may need helper method: getObjectProperty($obj, $property), this this messy code can be:

```php
<?php

$client = $this->getObjectProperty($obj, 'client');
$options = $this->getObjectProperty($client, 'defaultOptions');
```